src/zope/formlib/interfaces.py

```diff
@@ -328,17 +328,15 @@ class ITextBrowserWidget(ISimpleInputWidget):
             u'If True, an empty string is converted to field.missing_value.'),
         default=True)
 
-# XXX this seems to be buggy (since at least 2005)
-# it returns None, and prefix_re is never defined.
-
 
 def reConstraint(pat, explanation):
     pat = re.compile(pat)
 
     def constraint(value):
-        if prefix_re.match(value):  # noqa: F821 undefined name
+        if pat.match(value):
             return True
         raise Invalid(value, explanation)
+    return constraint
```

Comment:
A unit test wouldn't be amiss?

Reply:
Added in https://github.com/zopefoundation/zope.formlib/pull/35/commits/1c84b39469ab126d4699c875b003f480d5246a1e.